### PR TITLE
Fix path tests windows

### PR DESCRIPTION
**Pull request by robi-wan**
Prior this commit(s) three tests failed under windows.

### CODE REVIEW

File: lib/elixir/test/elixir/path_test.exs [hunk 1]
```diff
@@ -45,8 +45,10 @@ defmodule PathTest do
       assert Path.relative_to("D:/usr/local/foo", "D:/usr/") == "local/foo"
       assert Path.relative_to("D:/usr/local/foo", "d:/usr/") == "local/foo"
       assert Path.relative_to("d:/usr/local/foo", "D:/usr/") == "local/foo"
-      assert Path.relative_to("D:/usr/local/foo", "d:") == "usr/local/foo"
-      assert Path.relative_to("D:/usr/local/foo", "D:") == "usr/local/foo"
+      assert Path.relative_to("D:/usr/local/foo", "d:/") == "usr/local/foo"
+      assert Path.relative_to("D:/usr/local/foo", "D:/") == "usr/local/foo"
+      assert Path.relative_to("D:/usr/local/foo", "d:") == "D:/usr/local/foo"
+      assert Path.relative_to("D:/usr/local/foo", "D:") == "D:/usr/local/foo"
```
Comment (eksperimental) [inline lib/elixir/test/elixir/path_test.exs]: these tests will not pass on Linux

Comment (eksperimental) [inline lib/elixir/test/elixir/path_test.exs]: something needs to be fixed in the function for Windows

Comment (robi-wan) [inline lib/elixir/test/elixir/path_test.exs]: These tests are just executed [under windows](https://github.com/robi-wan/elixir/blob/8997b632d3721f7e685b98be45f5e7e4ce946611/lib/elixir/test/elixir/path_test.exs#L35-L36):

    if windows? do
      test "relative win" do

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

File: lib/elixir/test/elixir/path_test.exs (hunk 1 above)
Comment (eksperimental) [inline lib/elixir/test/elixir/path_test.exs]: so sorry i didn't read the context

**Pull request merged**